src/components/LanguagePicker.tsx

```diff
@@ -0,0 +1,456 @@
+                            onClick={onMenuClose}
+                          >
+                            <Text fontSize="lg" color="primary.base">
+                              {targetName}
+                            </Text>
+                            <Text
+                              textTransform="uppercase"
+                              fontSize="xs"
+                              color="body.medium"
+                            >
+                              {sourceName}
+                            </Text>
+                            <Text
+                              textTransform="lowercase"
+                              fontSize="xs"
+                              color="body.medium"
+                              maxW="full"
+                            >
+                              {progress} translated • {words} words
+                            </Text>
+                            <Progress value={approvalProgress} />
+                          </Item>
+                        )
+                      }
+                    )}
+                    <MenuDivider borderColor="body.medium" my="4" mx="-2" />
+                  </>
+                )}
+
+                <Text fontSize="xs" color="body.medium">
+                  Filter list ({filteredNames.length} languages)
+                </Text>
+                <MenuItem
+                  onFocus={() => inputRef.current?.focus()}
+                  p="0"
+                  bg="transparent"
+                  position="relative"
+                  closeOnSelect={false}
+                >
+                  <Input
+                    placeholder="Type to filter"
+                    value={filterValue}
+                    onChange={(e) => setFilterValue(e.target.value)}
+                    ref={inputRef}
+                    h="8"
+                    mt="1"
+                    mb="2"
+                    bg="background.base"
+                    color="body.base"
+                    onKeyDown={(e) => {
+                      // Navigate to first result on enter
+                      if (e.key === "Enter") {
+                        e.preventDefault()
+                        firstItemRef.current?.click()
+                      }
+                    }}
+                  />
+                </MenuItem>
+
+                {filteredNames.map(
+                  (
+                    {
+                      localeOption,
+                      sourceName,
+                      targetName,
+                      approvalProgress,
+                      wordsApproved,
+                    },
+                    index
+                  ) => {
+                    const firstResult = index === 0
+                    const { progress, words } = getProgressInfo(
+                      approvalProgress,
+                      wordsApproved
+                    )
+                    return (
+                      <Item
```

Comment:
We could have this whole return as a separate component inside the `LanguagePicker` folder, called something like `LanguagePickerItem.tsx` that handles all the display logic for a given item. That could be reuse in L311 (browser language item) too.

Reply:
Hah, yeah I made an initial attempt at this by extracting the `Item` component at the top of this file. I didn't abstract the `children` though, which would be helpful here. Will abstract this all into another file to clean it all up, thanks!